lib/dolly/mango/selector.rb

```diff
@@ -0,0 +1,147 @@
+module Dolly
+  module Mango
+    module Selector
+
+      SELECTOR     = 'selector'.freeze
+
+      # Equality Operators
+      EQ_OPERATOR  = '$eq'.freeze
+      NE_OPERATOR  = '$ne'.freeze
+      GT_OPERATOR  = '$gt'.freeze
+      LT_OPERATOR  = '$lt'.freeze
+      GTE_OPERATOR = '$gte'.freeze
+      LTE_OPERATOR = '$lte'.freeze
+
+      EQUALITY_OPERATORS = [
+                             EQ_OPERATOR,
+                             NE_OPERATOR,
+                             GT_OPERATOR,
+                             LT_OPERATOR,
+                             GTE_OPERATOR,
+                             LTE_OPERATOR,
+                           ].freeze
+
+      # Object Operators
+      EXISTS_OPERATOR = '$exists'.freeze
+      TYPE_OPERATOR   = '$type'.freeze
+
+      OBJECT_OPERATORS = [EXISTS_OPERATOR, TYPE_OPERATOR].freeze
+
+      # Array Operators
+      IN_OPERATOR  = '$in'.freeze
+      NIN_OPERATOR = '$nin'.freeze
+      SIZE_OPERATOR = '$size'.freeze
+
+      ARRAY_OPERATORS = [IN_OPERATOR, NIN_OPERATOR].freeze
+
+      # Miscellaneous Operators
+      MOD_OPERATOR   = '$mod'.freeze
+      REGEX_OPERATOR = '$regex'.freeze
+
+      MISC_OPERATORS = [MOD_OPERATOR, REGEX_OPERATOR].freeze
+
+      # Combination Operators
+      OR_OPERATOR  = '$or'.freeze
+      NOT_SELECTOR = '$not'.freeze
+      NOR_SELECTOR = '$nor'.freeze
+      AND_OPERATOR = '$and'.freeze
+      ALL_OPERATOR = '$all'.freeze
+      EM_OPERATOR  = '$elemMatch'.freeze
+
+      COMBINATION_OPERATORS = [
+                                OR_OPERATOR,
+                                NOT_SELECTOR,
+                                NOR_SELECTOR,
+                                AND_OPERATOR,
+                                ALL_OPERATOR,
+                                EM_OPERATOR
+                              ]
+
+      POSSIBLE_TYPE_VALUES = %w/null boolean number string array object/.freeze
+
+      ALL_OPERATORS = [
```

Comment:
think you could drop the `.freeze`s from all of the constant declarations and freeze them in place using the `ALL_OPERATORS` array. think it would work....

``` ruby
LTE_OPERATOR = '$lte'
LTE_OPERATOR.frozen? => false
[LTE_OPERATOR].map!(&:freeze)
LTE_OPERATOR.frozen? => true
```

``` ruby
ALL_OPERATORS.flatten.map!(&:freeze)
ALL_OPERATORS.freeze
```